### PR TITLE
Update dependencies, analysis and fix WidgetsBinding.instance

### PR DESCRIPTION
**Pull request by jocubeit**
A series of commits that updates this excellent package to the recent stable Flutter and new analysis using flutter_lints.

Also bumped the Version to 0.4.0.

Tests all pass.

Hope you can merge and publish this soon, I'd like to use it in a production release.

### CODE REVIEW

**Comment by TNorbury** [comment]
Thanks for the PR, I'll take a look at this in the next few days!

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ In your Flutter project, add the package to your dependencies
 
 ```yml
 dependencies:
-    reorderable_carousel: ^0.3.1
+    reorderable_carousel: ^0.4.0
```
Comment (TNorbury) [inline README.md]: I have some other things I want to do before rolling out this release, so I'll handle all the version stuff
```suggestion
    reorderable_carousel: ^0.3.1
```

File: analysis_options.yaml [hunk 2]
```diff
@@ -1,8 +1,37 @@
...
+  # and their documentation is published at
+  # https://dart-lang.github.io/linter/lints/index.html.
+  #
+  # Instead of disabling a lint rule for the entire project in the
+  # section below, it can also be suppressed for a single line of code
+  # or a specific dart file by using the `// ignore: name_of_lint` and
+  # `// ignore_for_file: name_of_lint` syntax on the line or in the file
+  # producing the lint.
+  rules:
+    implementation_imports: false
```
Comment (TNorbury) [inline analysis_options.yaml:25]: please add back the removed rules
```suggestion
    implementation_imports: false
    constant_identifier_names: false
    omit_local_variable_types: false
    avoid_types_on_closure_parameters: false
    use_setters_to_change_properties: false
```

File: pubspec.yaml [hunk 3]
```diff
@@ -1,21 +1,20 @@
 name: reorderable_carousel
 description: A carousel in that allows one to reorder the contents of it via drag and drop 
-version: 0.3.2+1
+version: 0.4.0
```
Comment (TNorbury) [inline pubspec.yaml]: I'll take care of the versioning
```suggestion
version: 0.3.2+1
```

File: pubspec.yaml [hunk 4]
```diff
@@ -1,21 +1,20 @@
...
 homepage: https://github.com/TNorbury/reorderable_carousel
 repository: https://github.com/TNorbury/reorderable_carousel
 issue_tracker: https://github.com/TNorbury/reorderable_carousel/issues
 
 environment:
-  sdk: ">=2.12.0 <3.0.0"
-  flutter: ">=2.0.0"
+  sdk: ">=2.17.0 <3.0.0"
```
Comment (TNorbury) [inline pubspec.yaml:9]: This needs to stay in here to indicate that Flutter is required in order to use this package
```suggestion
  sdk: ">=2.17.0 <3.0.0"
  flutter: ">=2.0.0"
```

File: pubspec.yaml [hunk 5]
```diff
@@ -1,21 +1,20 @@
...
 
 dependencies:
   flutter:
     sdk: flutter
 
 dev_dependencies:
-  effective_dart: ^1.3.0
   flutter_test:
     sdk: flutter
+  flutter_lints:  ^2.0.0
```
Comment (TNorbury) [inline pubspec.yaml]: alphabetize :)
```suggestion
  flutter_lints:  ^2.0.0
  flutter_test:
    sdk: flutter
```

**Review by TNorbury — CHANGES_REQUESTED** [review]
Thanks for your work, requested a few changes

File: pubspec.yaml (hunk 4 above)
Comment (TNorbury) [reply pubspec.yaml:9]: I was incorrect here, it can be left as is.

**Comment by jocubeit** [comment]
Thanks for reviewing and suggesting changes. All requests resolved.

**Pull request merged**